smartsim/_core/_cli/build.py

```diff
@@ -331,7 +346,7 @@ def configure_parser(parser: argparse.ArgumentParser) -> None:
         "--device",
         type=str.lower,
         default=Device.CPU.value,
-        choices=[*(device.value for device in Device), "gpu"],
+        choices=available_devices,
```

Comment:
Another sanity check: do we want to explicitly disallow `--device=gpu` as an option?